**Sanger Sequencing** 

This one is widely taught and known, but a little outdated. 

**1.** Lyse the cells and extract DNA. The DNA itself is fragmented and copied many many times over. 

**2.** Attach primers to the fragments and separate the experiment into 4 tubes. 

**3.** Each tube receives plenty DNA polymerase, plenty deoxynucleosidetriphosphates (dNTPs) and about 1/100th the amount of di-deoxynucleotidetriphosphates (ddNTPs). The issue with ddNTPs is that they don't have the 3'-OH group. Synthesis stops when it a ddNTP is attached, which happens at various points on various fragments. 

**4.** DNA is negatively charged. Heat the DNA to separate it and put through a gel from pos->neg side, which separates fragments with a resolution of 1 nucleotide. 

**5.** The trick is that the ddNTPs are fluorescent! You now have many many fragments whose lengths you know down to the nucleotide. Expose an x-ray film to the gel, and now you have 4 rows representing the 4 nucleotides and dark bands where each of the 4 nucleotides terminated a fragment. Here's a picture to clarify: 

![image of sanger gel](https://upload.wikimedia.org/wikipedia/commons/c/cb/Sequencing.jpg)

**PacBio Sequencing:** 

**1.** DNA is immobilized at the bottom of a small well with a DNA polymerase. 

**2.** Fluorescent nucleotides are introduced to into the well, each labeled with a unique fluorophore.

**3.** Light from the bottom of the well makes the bases fluoresce as they are added to the DNA template by the polymerase and the sequence of light colors is recorded.

